src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops.cpp

```diff
@@ -0,0 +1,1074 @@
+    const auto& lowp_quant = conv.lowp_quant;
+    result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle, const boost::any& primitive_params) {
+            const auto invoke_params = boost::any_cast<conv::WrWInvokeParams>(primitive_params);
+            const auto& tensors      = invoke_params.tensors;
+            auto kernel              = handle.Run(kernels[0]);
+            float elapsed            = 0;
+            float zero               = 0.f;
+            if(ctx.IsFp16() || ctx.IsBfp16())
+            {
+                const auto& workSpace = invoke_params.workSpace;
+                TensorDescriptor workspaceDesc(
+                    miopenFloat, tensors.dwDesc.GetLengths(), tensors.dwDesc.GetStrides());
+                SetTensor(handle, workspaceDesc, workSpace, &zero);
+                if(handle.IsProfilingEnabled())
+                {
+                    elapsed += handle.GetKernelTime();
+                }
+                kernel(tensors.x, tensors.dy, workSpace);
+                if(handle.IsProfilingEnabled())
+                {
+                    elapsed += handle.GetKernelTime();
+                }
+                CastTensor(handle,
+                           &lowp_quant,
+                           workspaceDesc,
+                           workSpace,
+                           tensors.dwDesc,
+                           tensors.dw,
+                           0,
+                           0);
+            }
+            else
+            {
+                SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
+                handle.Run(kernels[0])(tensors.x, tensors.dy, tensors.dw);
+            }
+
+            if(handle.IsProfilingEnabled())
+            {
+                elapsed += handle.GetKernelTime();
+                handle.ResetKernelTime();
+                handle.AccumKernelTime(elapsed);
+            }
+        };
+    };
+    result.workspce_sz = GetWorkspaceSize(ctx);
+    return result;
+}
+
+int ConvHipImplicitGemmWrwV4R4Xdlops::RunAndMeasureSolution(const miopen::Handle& profile_h,
+                                                            ConstData_t bot_buf,
+                                                            ConstData_t top_buf,
+                                                            Data_t wei_buf,
+                                                            ConstData_t bias_buf,
+                                                            const ConvolutionContext& ctx,
+                                                            const ConvSolution& solution,
+                                                            float& elapsed_time) const
+{
+    assert(bias_buf == nullptr);
+    (void)bias_buf;
+
+    return RunAndMeasureSolutionBase(
+        profile_h, bot_buf, top_buf, wei_buf, ctx, solution, elapsed_time);
+}
+
+bool ConvHipImplicitGemmWrwV4R4Xdlops::IsApplicable(const ConvolutionContext& ctx) const
```

Comment:
It it applicable for gfx8? gfx900? gfx906?

Reply:
This is for gfx908. The hardware is detected inside ```IsXdlopsSupport()``` below, so no explicit detection is needed here

Reply:
This is for gfx908. The hardware is detected inside IsXdlopsSupport() below, so no explicit detection is needed here

Reply:
Ok